crates/fs/src/fs.rs

```diff
@@ -1337,6 +1336,34 @@ pub fn copy_recursive<'a>(
     .boxed()
 }
 
+#[cfg(target_os = "windows")]
+fn file_id(path: impl AsRef<Path>) -> u64 {
+    use std::os::windows::{fs::OpenOptionsExt, io::AsRawHandle};
+
+    use windows_sys::Win32::{
+        Foundation::HANDLE,
+        Storage::FileSystem::{
+            GetFileInformationByHandle, BY_HANDLE_FILE_INFORMATION, FILE_FLAG_BACKUP_SEMANTICS,
+        },
+    };
+
+    let file = std::fs::OpenOptions::new()
```

Comment:
We shouldn't be doing synchronous file opening here, this needs to be swapped to use `smol::fs`